### ISSUE TITLE
Allow role=combobox on more input types

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1777]
The following should pass `aria-allowed-role`:

```html
<input type="search" 
  role="combobox"
  aria-expanded="false"
  aria-label="search" />
```

This is allowed according to the ARIA 1.0 spec, but it isn't allowed by HTML-ARIA. Given their status, I think the ARIA spec should get priority. Since the following are `combobox` by default if you put a `list` attribute on them, I think axe-core should allow the following types to be a combobox:

- text
- search
- tel
- url
- email

**Comment by WilcoFiers** [comment]
That's actually the same as what we have here, making that consistent as well:
https://github.com/dequelabs/axe-core/blob/develop/lib/checks/aria/required-children.js#L67

**Comment by straker** [comment]
Just need to update the combobox [`allowedElements` object](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/aria/index.js#L423-L448) in our master aria file to allow an array of `types` instead of a single type.

```js
nodeName: 'input',
properties: {
  type: ['text', 'search', 'tel', 'url', 'email']
}
```

Then update the [integration test for aria-allowed-role](https://github.com/dequelabs/axe-core/blob/develop/test/integration/rules/aria-allowed-role/aria-allowed-role.html#L41) to include inputs of these types using `role="combobox"`. The integration file is a bit odd in that you create an element that should pass the rule and give it a unique id. Then you'll need to update the [associated integration json](https://github.com/dequelabs/axe-core/blob/develop/test/integration/rules/aria-allowed-role/aria-allowed-role.json#L4-L55) `passes` array to include each of the newly added ids in their own array.

**Comment by jeankaplansky** [comment]
@straker: This list is final? No other type values than these?:

- text
- search
- tel
- url
- email

Please confirm. Thanks!

**Comment by straker** [comment]
@jeankaplansky Should be

**Comment by padmavemulapati** [comment]
Verified with latest axe-coconut and axe, looks everything working fine
